.github/workflows/test.yaml

```diff
@@ -800,18 +860,21 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 7
+
       - uses: actions/setup-node@v3
         with:
-          node-version: 14
-          cache: 'yarn'
-          # Only cache the root node_modules
-          cache-dependency-path: 'yarn.lock'
+          node-version: 16
+          cache: ${{ contains(matrix.platform, 'azure') && 'npm' || 'pnpm' }}
+          cache-dependency-path: ${{ github.job }}/${{ matrix.platform }}/${{ contains(matrix.platform, 'azure') && 'package-lock.json' || 'pnpm-lock.yaml' }}
```

Comment:
?

Reply:
There's not ternaries in workflow syntax so this is conditional cache selection

Reply:
It improved job speed

Reply:
So it hard codes `package-lock.json` for any projects containing `azure` and `pnpm-lock.yaml` for all others?